pkg/api/diagnostic_test.go

```diff
@@ -0,0 +1,56 @@
+package api_test
+
+import (
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/wakatime/wakatime-cli/pkg/api"
+	"github.com/wakatime/wakatime-cli/pkg/diagnostic"
+	"github.com/wakatime/wakatime-cli/pkg/version"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_SendDiagnostics(t *testing.T) {
+	url, router, close := setupTestServer()
+	defer close()
+
+	var numCalls int
+
+	router.HandleFunc("/v1/plugins/errors", func(w http.ResponseWriter, req *http.Request) {
+		numCalls++
+
+		// check method and headers
+		assert.Equal(t, http.MethodPost, req.Method)
+		assert.Equal(t, []string{"application/json"}, req.Header["Content-Type"])
+
+		// check body
+		expectedBody, err := ioutil.ReadFile("testdata/diagnostics_request.json")
```

Comment:
should we test to make sure `Authentication` header isn't present?

Reply:
Makes sense, but the test in `api` package initializes the api client itself. Only a test in `./cmd/legacy` will give full assurance, as this is the place, where the client is initialized in real execution. Will add it :+1:

Reply:
@alanhamlett Check latest commit, which disables auth for diagnostics and verifies it with a test.